components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -1152,6 +1154,58 @@ public static String generateSecretKey(String channel, String tenantDomain, Stri
         }
     }
 
+    /**
+     * Generate a secret key according to the given channel. Method will generate an OTP for mobile channel, and an OTP
+     * for e-mail channel when 'sendOTPInEmail' property enabled. For other scenarios method will generate UUID.
+     *
+     * @param channel Recovery notification channel.
+     * @param recoveryScenario       Recovery scenario.
+     * @param tenantDomain           Tenant domain.
+     * @param connectorName          Connector name.
+     * @return Secret key.
+     * @throws IdentityRecoveryServerException while getting OTP generated.
+     */
+    public static String generateSecretKey(String channel, String recoveryScenario, String tenantDomain, String connectorName)
+            throws IdentityRecoveryServerException {
+
+        // Set default OTP configuration values, for scenarios that don't have specific OTP configurations.
+        boolean useUppercase = true;
+        boolean useLowercase = true;
+        boolean useNumeric = true;
+        boolean sendOTPInEmail = false;
+        int otpLength = IdentityRecoveryConstants.OTP_CODE_DEFAULT_LENGTH;
+        // Set connector specific OTP configuration values, for connectors that have separate OTP configurations.
+        if (StringUtils.isNotBlank(connectorName)) {
+            sendOTPInEmail = Boolean.parseBoolean(getRecoveryConfigs(
+                    connectorName + ".OTP.SendOTPInEmail", tenantDomain));
```

Comment:
Shall we define `.OTP.SendOTPInEmail` as constant and use here if they are used in multiple places ? check below similar lines as well.